lib/Transport.js

```diff
@@ -1,21 +1,44 @@
 "use strict";
-var Transport = module.exports = function(options) {
-  options = options || { };
-  this._construct(options);
+var Transport = module.exports = function(path, auth) {
+  path = path || { };
+  auth = auth || { };
+  this._construct(path, auth);
 };
 
 var request = require("superagent");
 
-Transport.prototype._construct = function(path) {
+Transport.prototype._construct = function(path, auth) {
   this._path = path;
+  this._auth = {
+    cookie: auth.cookie,
+    header: auth.header
+  };
+};
+
+Transport.prototype._attachAuthToRequest = function(someRequest) {
+  if (this._auth.cookie) {
+    var cookieName = Object.keys(this._auth.cookie)[0];
+    someRequest.set("Cookie", cookieName + "=" + this._auth.cookie[cookieName]);
+  }
+
+  if (this._auth.header) {
+    var headerName = Object.keys(this._auth.header)[0];
+    someRequest.set(headerName, this._auth.header[headerName]);
+  }
 };
 
 Transport.prototype._action = function(method, url, data, callback) {
   // console.log(method, url, JSON.stringify(data, null, 2));
-  request[method](url).send(data).end(function(err, payload) {
+  var someRequest = request[method](url);
+  this._attachAuthToRequest(someRequest);
+  someRequest.send(data).end(function(err, payload) {
     // console.log("=>", err, JSON.stringify(JSON.parse(payload.text), null, 2));
     if (err) {
       // console.log(err.response.text);
+      if (err.status === 401) {
+        return callback(new Error("401 Unauthorized"));
```

Comment:
Will this possibly swallow useful context the server could give back. Does the jsonapi-server allow returning payloads, conceivably something like:
```
401 Not Authorised
{ "token_refresh_url": "http:/my-auth-server/" }
```